Libraries/Opc.Ua.Security.Certificates/Common/AsnUtils.cs

```diff
@@ -0,0 +1,160 @@
+/* ========================================================================
+ * Copyright (c) 2005-2020 The OPC Foundation, Inc. All rights reserved.
+ *
+ * OPC Foundation MIT License 1.00
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+using System;
+using System.Formats.Asn1;
+using System.Text;
+
+namespace Opc.Ua.Security.Certificates
+{
+    internal static class AsnUtils
+    {
+        /// <summary>
+        /// Converts a buffer to a hexadecimal string.
+        /// </summary>
+        internal static string ToHexString(this byte[] buffer, bool bigEndian = false)
+        {
+            if (buffer == null || buffer.Length == 0)
+            {
+                return String.Empty;
+            }
+
+            StringBuilder builder = new StringBuilder(buffer.Length * 2);
+
+            if (bigEndian)
+            {
+                for (int ii = buffer.Length - 1; ii >= 0; ii--)
+                {
+                    builder.AppendFormat("{0:X2}", buffer[ii]);
+                }
+            }
+            else
+            {
+                for (int ii = 0; ii < buffer.Length; ii++)
+                {
+                    builder.AppendFormat("{0:X2}", buffer[ii]);
+                }
+            }
```

Comment:
In Big Endian the number 64,208 (0xFACE) is written as FA CE.  In Little Endian it's written as CE FA.

Unless the data in your buffer is already reversed to Little Endian order, the name of the boolean and the functionality it provides seem reversed.